scripts/variantstore/wdl/GvsCreateVATAnnotations.wdl

```diff
@@ -0,0 +1,320 @@
+    # while extracting and calculating the an/ac/af & sc by subpopulation into a tsv
+    command <<<
+        set -e
+
+        if [ ~{has_service_account_file} = 'true' ]; then
+          gsutil cp ~{service_account_json_path} local.service_account.json
+          export GOOGLE_APPLICATION_CREDENTIALS=local.service_account.json
+          gcloud auth activate-service-account --key-file=local.service_account.json
+
+        fi
+
+        cp ~{custom_annotations_template} ~{custom_annotations_file_name}
+
+        gsutil cp ~{input_vcf} ~{local_input_vcf}
+        gsutil cp ~{input_vcf_index} ~{local_input_vcf_index}
+        gsutil cp ~{ref} Homo_sapiens_assembly38.fasta
+
+        ## compare the ancestry sample list with the vcf sample list
+        # TODO throw an error (but dont fail the job) if there are samples that are in one, but not the other. Throw two different errors.
+        # Currently commented out to save on io with the AoU beta VAT creation
+        # awk '{print $1}' ~{subpopulation_sample_list} | tail -n +2 | sort -u > collected_subpopulation_samples.txt
+        # bcftools query --list-samples ~{local_input_vcf} | sort -u > collected_samples.txt
+        # diff collected_subpopulation_samples.txt collected_samples.txt
+
+        # expected_subpopulations = [
+        # "afr",
+        # "amr",
+        # "eas",
+        # "eur",
+        # "mid",
+        # "oth",
+        # "sas"
+        #]
+
+        ## track the dropped variants with +50 alt alleles or N's in the reference (Since Nirvana cant handle N as a base, drop them for now)
+        bcftools view -i 'N_ALT>50 || REF~"N"' ~{local_input_vcf} | bcftools query -f '%CHROM\t%POS\t%ID\t%REF\t%ALT\n' > track_dropped.tsv
+
+        wc -l track_dropped.tsv
+
+        ## filter out sites with too many alt alleles
+        bcftools view -e 'N_ALT>50 || REF~"N"' --no-update ~{local_input_vcf} -Ou | \
+        ## filter out the non-passing sites
+        bcftools view  -f 'PASS,.' --no-update -Oz -o filtered.vcf.gz
+        ## normalize, left align and split multi allelic sites to new lines, remove duplicate lines
+        bcftools norm -m- --check-ref w -f Homo_sapiens_assembly38.fasta filtered.vcf.gz -Oz -o normalized.vcf.gz
+        rm ~{local_input_vcf}
+        ## filter out spanning deletions and variants with an AC of 0
+        bcftools view -e 'ALT[0]="*" || AC=0' --no-update normalized.vcf.gz -Ou | \
+        ## ensure that we respect the FT tag
+        bcftools filter -i "FORMAT/FT='PASS,.'" --set-GTs . -Oz -o ~{normalized_vcf}
+
+        ## clean up unneeded file
+        rm normalized.vcf.gz
+
+        ## During normalization, sometimes duplicate variamts appear but with different calculations. This seems to be a bug in bcftools. For now we are dropping all duplicate variants
+        ## The way in which this is done is a bit hamfisted and should be optimized in the future.
```

Comment:
like a TODO, take out in favor of creating a ticket; we can always decide "won't do"